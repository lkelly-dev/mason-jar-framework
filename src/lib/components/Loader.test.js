import React from 'react';
import { shallow } from 'enzyme';
import { Loader } from "../index.js";

describe('Loader', () => {
  it('Loader matches snapshot', () => {
    const component = shallow(<Loader></Loader>);
    expect(component).toMatchSnapshot();
  });

  it('has expected props', () => {
    const component = shallow(
      <Loader
        size={ size }
        color={ color }
        className={ cn }
        what="ever" />
    );
    const classNameArray = component.prop('className').split(' ');
    expect(classNameArray).toContain(cn);
    expect(classNameArray).toContain('loader');
    expect(classNameArray).toContain(`loader-${size}`);
    expect(classNameArray).toContain(`loader-${color}`);
    expect(classNameArray).toHaveLength(4);
    expect(component.prop('what')).toBe('ever');
  });
  it('doesnt include invalid props', () => {
    const component = shallow(
      <Loader size={ badSize } color={ badColor } />
    );
    const classNameArray = component.prop('className').split(' ');
    expect(classNameArray).toContain('loader');
    expect(classNameArray).not.toContain(`loader-${badSize}`);
    expect(classNameArray).not.toContain(`loader-${badColor}`);
  });
});
