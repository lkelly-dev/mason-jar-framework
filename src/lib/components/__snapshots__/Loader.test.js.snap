// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader Loader matches snapshot 1`] = `
<div
  className="loader"
/>
`;
